### PR TITLE
Support NeoPixel RGB

### PR DESCRIPTION
**Pull request by andyp1per**
NeoPixels can take either RGB or GRB order, with GRB being the default. This adds support for RGB order.

This work was sponsored by Proto Systems

### CODE REVIEW

File: libraries/AP_SerialLED/AP_SerialLED.h [hunk 1]
```diff
@@ -32,15 +33,15 @@
 class AP_SerialLED {
 public:
     // set number of LEDs per pin
-    bool set_num_neopixel(uint8_t chan, uint8_t num_leds);
+    bool set_num_neopixel(uint8_t chan, uint8_t num_leds, bool rgb = false);
```
Comment (IamPete1) [inline libraries/AP_SerialLED/AP_SerialLED.h:36]: I would prefer a new method so we can expose it to scripting. We don't support optional arguments for auto generated bindings. `set_num_neopixel_rgb` sort of thing.

**Comment by IamPete1** [comment]
This ticks off one of the color sequences from https://github.com/ArduPilot/ardupilot/issues/13369

File: libraries/AP_SerialLED/AP_SerialLED.h (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_SerialLED/AP_SerialLED.h:36]: the issue is the setup is done at the same time, do the bindings support overloading? Or do you mean something like ```set_num_neopixel_rgb()```

**Comment by andyp1per** [comment]
RGBW seems much harder since we will have to change all of the output code.

File: libraries/AP_SerialLED/AP_SerialLED.h (hunk 1 above)
Comment (IamPete1) [reply libraries/AP_SerialLED/AP_SerialLED.h:36]: I was just thinking a new method name so we then have three, our led scripts typically looks like this:

https://github.com/ArduPilot/ardupilot/blob/1641aa42ec37884caf204195c5d97eb97663e334/libraries/AP_Scripting/examples/LED_roll.lua#L29-L31

Where you would just on-comment the type you have.

Comment (andyp1per) [reply libraries/AP_SerialLED/AP_SerialLED.h]: Fixed

**Review by tridge — APPROVED** [review]

**Pull request merged**